src/test/optimizer/strategy/or_to_union_rule_test.cpp

```diff
@@ -0,0 +1,219 @@
+    b_b = node_b->get_column("b");
+
+    node_c = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "a"}, {DataType::Int, "b"}, {DataType::Int, "c"}}, "c");
+    c_a = node_c->get_column("a");
+    c_b = node_c->get_column("b");
+
+    customer_demographics =
+        MockNode::make(MockNode::ColumnDefinitions{{DataType::Int, "c_customer_sk"}}, "customer_demographics");
+    c_customer_sk = customer_demographics->get_column("c_customer_sk");
+
+    date_dim = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "d_date_sk"}, {DataType::Int, "d_year"}, {DataType::Int, "d_qoy"}},
+        "date_dim");
+    d_date_sk = date_dim->get_column("d_date_sk");
+    d_year = date_dim->get_column("d_year");
+    d_qoy = date_dim->get_column("d_qoy");
+
+    web_sales = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "ws_sold_date_sk"}, {DataType::Int, "ws_bill_customer_sk"}},
+        "web_sales");
+    ws_sold_date_sk = web_sales->get_column("ws_sold_date_sk");
+    ws_bill_customer_sk = web_sales->get_column("ws_bill_customer_sk");
+
+    catalog_sales = MockNode::make(
+        MockNode::ColumnDefinitions{{DataType::Int, "cs_sold_date_sk"}, {DataType::Int, "cs_ship_customer_sk"}},
+        "catalog_sales");
+    cs_sold_date_sk = catalog_sales->get_column("cs_sold_date_sk");
+    cs_ship_customer_sk = catalog_sales->get_column("cs_ship_customer_sk");
+
+    _rule = std::make_shared<OrToUnionRule>();
+  }
+
+  std::shared_ptr<OrToUnionRule> _rule;
+
+  std::shared_ptr<MockNode> node_a, node_b, node_c, customer_demographics, date_dim, web_sales,
+      catalog_sales;
+  LQPColumnReference a_a, a_b, b_a, b_b, c_a, c_b, c_customer_sk, d_date_sk, d_year, d_qoy, ws_sold_date_sk,
+  ws_bill_customer_sk, cs_sold_date_sk, cs_ship_customer_sk;
+};
+
+TEST_F(OrToUnionRuleTest, TwoExistsToUnion) {
+  // SELECT * FROM a WHERE EXISTS (SELECT * FROM b WHERE b.b = a.b) OR EXISTS (SELECT * FROM c WHERE c.b = a.b)
+
+  const auto parameter = correlated_parameter_(ParameterID{0}, a_a);
+
+  // clang-format off
+  const auto subquery_lqp_a =
+  PredicateNode::make(equals_(b_b, parameter),
+    node_b);
+
+  const auto subquery_a = lqp_subquery_(subquery_lqp_a, std::make_pair(ParameterID{0}, a_b));
+
+  const auto subquery_lqp_b =
+  PredicateNode::make(equals_(c_b, parameter),
+    node_c);
+
+  const auto subquery_b = lqp_subquery_(subquery_lqp_b, std::make_pair(ParameterID{0}, a_b));
+
+  const auto input_lqp =
+  PredicateNode::make(or_(exists_(subquery_a), exists_(subquery_b)),
+    node_a);
+
+  const auto expected_lqp =
+  UnionNode::make(UnionMode::Positions,
+    PredicateNode::make(exists_(subquery_a),
+      node_a),
+    PredicateNode::make(exists_(subquery_b),
+      node_a));
+  // clang-format on
+
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, input_lqp);
+
+  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
+}
+
+TEST_F(OrToUnionRuleTest, NoRewriteThreeExistsToUnion) {
```

Comment:
What are we exactly testing here? We should test that (a) the new code does what it advertises to do and (b) that it does not break anything, neither now, nor in what we can predict for the future. We should not test that it does not do more than advertised - that just leads to confusion when the feature is implemented.